pkg/controller/clusterprovision/clusterprovision_controller_test.go

```diff
@@ -102,12 +102,13 @@ func TestClusterProvisionReconcile(t *testing.T) {
 			expectedStage: hivev1.ClusterProvisionStageInitializing,
 		},
 		{
-			name: "completed job",
+			name: "removed completed job after successful install",
```

Comment:
Keep the old name. The test is doing more than ensuring that the completed job is removed. It is also validating that the stage is transitioned correctly.